sklearn/model_selection/_validation.py

```diff
@@ -747,6 +768,8 @@ def _fit_and_score(
         result["parameters"] = parameters
     if return_estimator:
         result["estimator"] = estimator
+    if return_indices:
+        result["indices"] = (train, test)
```

Comment:
Since the `train` and `test` are already known at the call site, I do not think `_fit_and_score` needs to return this data back. Also returning a large ndarray can be more costly due to serialization from loky-backed multiprocessing.

If `return_indices=True`, can the caller `cross_validate` construct the indices before handing them off to the `Parallel` call? Concretely:

```python
indices = iter(cv.split(X, y, groups))
if return_indices:
    # Materialize the indices to be returned
    indices = list(indices)

results = parallel(...)(... for train, test in indices)
```

Reply:
Indeed, it makes sense. Materializing is not an issue here since we need to be materialized to store it.